Framework/API/src/ExperimentInfo.cpp

```diff
@@ -219,7 +217,8 @@ void ExperimentInfo::setInstrument(const Instrument_const_sptr &instr) {
   // instrument may now suddenly be valid, so we have to reinitialize the
   // detector grouping. Also the index corresponding to specific IDs may have
   // changed.
-  invalidateSpectrumDefinitions();
+  std::fill(m_spectrumDefinitionNeedsUpdate.begin(),
```

Comment:
I'm not a big fan of how `ExperimentInfo` gets created and then setup like this, but I guess that's all part of a much larger issue that would be hard to change at this point.